sqlglot/dialects/redshift.py

```diff
@@ -169,6 +170,7 @@ class Generator(Postgres.Generator):
 
         TRANSFORMS = {
             **Postgres.Generator.TRANSFORMS,
+            exp.Array: rename_func("ARRAY"),
```

Comment:
I recall noticing this issue in the past, but IIRC redshift supports both `ARRAY(...)` and `ARRAY[...]` and they mean different things. So there's some ambiguity here where you lose either syntax if you keep a single representation, mapping's not 1-1. Can you double check this?

Reply:
Hmm, I couldn't get `ARRAY_CONCAT` to work with the bracket version and I can't seem to find it documented anywhere yet, I'll keep searching

Reply:
This is interesting, I've been searching the Redshift [data types](https://docs.aws.amazon.com/redshift/latest/dg/c_Supported_data_types.html), I can't find a native array type except `SUPER`:

```
SUPER
A superset data type that encompasses all scalar types of Amazon Redshift including 
complex types such as ARRAY and STRUCTS.
```


And while array functions don't work with the bracket version, it seems that queries like `SELECT ARRAY[1, 2]` can still go through, notice the different symbols though:

```SQL
SELECT ARRAY[1, 2, 1]
{1, 2, 1}

SELECT ARRAY(1, 2, 1)
[1, 2, 1]

SELECT ARRAY[1, 2, 1] = ARRAY(1, 2, 1);
ERROR: operator does not exist: integer[] = super

```

Lastly, this is from the AWS Postgres <-> Redshift [data type docs](https://docs.aws.amazon.com/redshift/latest/dg/federated-data-types.html):

```
The following RDS PostgreSQL and Aurora PostgreSQL data types are converted to VARCHAR(64K) in Amazon Redshift:

- JSON, JSONB
- Arrays
- ...
```

It might be that the bracket version is backwards-compatible Postgres and denotes a specific type of array, while the Redshift way is through `ARRAY()` which creates a `SUPER` type.

Reply:
In any case, we need to preserve `ARRAY[...]` and `ARRAY(...)` in redshift, right? This change is definitely breaking because it will produce the function syntax whereas right now we produce the square bracket one:

```python
>>> import sqlglot
>>> sqlglot.transpile("array(1,2,3)", "redshift")
['ARRAY[1, 2, 3]']
>>> sqlglot.transpile("array[1,2,3]", "redshift")
['ARRAY[1, 2, 3]']
```

Reply:
I think `ARRAY()` should be preserved here then, as this seems the preferred Redshift way according to docs, relevant SO threads etc. I didn't manage to find any source earlier showcasing the bracket version being used. Wdyt?

Reply:
I assume you mean we should represent `ARRAY(...)` using `exp.Array(...)` in redshift. If so, that's fine. What I'm saying is that we need to roundtrip both forms and that the current behavior is problematic because it changes the semantics. If you change `exp.Array` to generate `ARRAY(...)`, then this is gonna be a breaking change because as pointed above the current behavior is to generate the bracket syntax.